src/main/java/core/basesyntax/dao/StorageDaoImpl.java

```diff
@@ -10,7 +10,7 @@ public class StorageDaoImpl implements StorageDao {
     public User add(User user) {
         user.setId(++index);
```

Comment:
Using the pre-increment operator (++index) directly on the user's ID may not be thread-safe. If multiple threads access this method at the same time, it could lead to users being assigned the same ID. Consider using AtomicInteger for thread-safe operations or synchronizing the access to this method.